BootloaderCommonPkg/Library/ContainerLib/ContainerLib.c

```diff
@@ -412,37 +414,56 @@ AutheticateContainerInternal (
     }
   }
 
-  if (!EFI_ERROR (Status) && \
-      ((ContainerHdr->Flags & CONTAINER_HDR_FLAG_MONO_SIGNING) != 0)) {
-    // Additional verification if the container is signed monolithically.
-    // It is required for the container to be loaded in memory before registeration.
-    Status = EFI_UNSUPPORTED;
-    if ((ContainerHdr->Count > 1) && !IS_FLASH_ADDRESS (ContainerHeader)) {
-      // Use the last entry to verify all other combined components
-      CompEntry = (COMPONENT_ENTRY *)&ContainerHdr[1];
-      for (Index = 0; Index < (UINT32)(ContainerHdr->Count - 1); Index++) {
-        CompEntry = (COMPONENT_ENTRY *)((UINT8 *)(CompEntry + 1) + CompEntry->HashSize);
+  if (!EFI_ERROR (Status)) {
+    if ((ContainerHdr->Flags & CONTAINER_HDR_FLAG_MONO_SIGNING) != 0) {
+      // Additional verification if the container is signed monolithically.
+      // It is required for the container to be loaded in memory before registeration.
+      Status = EFI_UNSUPPORTED;
+      if ((ContainerHdr->Count > 1) && !IS_FLASH_ADDRESS (ContainerHeader)) {
+        // Use the last entry to verify all other combined components
+        CompEntry = (COMPONENT_ENTRY *)&ContainerHdr[1];
+        for (Index = 0; Index < (UINT32)(ContainerHdr->Count - 1); Index++) {
+          CompEntry = (COMPONENT_ENTRY *)((UINT8 *)(CompEntry + 1) + CompEntry->HashSize);
+        }
+        CompData    = (UINT8 *)(UINTN)(ContainerEntry->Base + ContainerHdr->DataOffset + CompEntry->Offset);
+        CompressHdr = (LOADER_COMPRESSED_HEADER *)CompData;
+        if (CompressHdr->Signature == LZDM_SIGNATURE) {
```

Comment:
Question: Does SG component is a compressed LZDM format?

Reply:
Yes, LZDM is "Dummy" compression type flag.